src/OpenLoco/src/Windows/CompanyList.cpp

```diff
@@ -1715,15 +1720,312 @@ namespace OpenLoco::Ui::Windows::CompanyList
+
+                    auto colour = self->getColour(WindowColour::secondary).c();
+                    auto paletteIndex = Colours::getShade(colour, remainder == 0 ? 6 : 4);
+
+                    drawingCtx.drawRect(xPos, gs.top, 1, height, paletteIndex, Gfx::RectFlags::none);
+                }
+
+                // No remainder means we get to draw a label on the horizontal axis, too
+                if (remainder == 0)
+                {
+                    int16_t xPos = ecx * gs.word_113DD80 + gs.left + gs.xOffset;
+                    int16_t yPos = gs.top + gs.height - gs.yOffset + 5;
+
+                    auto tr = Gfx::TextRenderer(drawingCtx);
+                    auto formatArgs = FormatArguments{};
+                    formatArgs.push(gs.xLabel);
+                    formatArgs.push(eax);
+
+                    tr.drawStringCentred({ xPos, yPos }, Colour::black, StringIds::graph_label_format, formatArgs);
+                }
+
+                eax += gs.xAxisStepSize;
+            }
+
+            // 0x004CFB5C
+            auto edx = 0;
+            while (true)
+            {
+                // TODO: can be moved down, around FormatArguments. Kept here for verification
+                int64_t ebx_eax = static_cast<int64_t>(edx) << gs.numValueShifts;
+
+                // Draw horizontal lines for each of the vertical axis labels
+                {
+                    auto colour = self->getColour(WindowColour::secondary).c();
+                    auto paletteIndex = Colours::getShade(colour, 6);
+
+                    auto xPos = gs.left + gs.xOffset - 2;
+                    auto width = gs.width - gs.xOffset + 3;
+                    auto yPos = -edx + gs.canvasHeight + gs.top;
+                    if (gs.flags & (1 << 0)) // never set
+                    {
+                        yPos -= gs.canvasHeight / 2;
+                    }
+
+                    drawingCtx.drawRect(xPos, yPos, width, 1, paletteIndex, Gfx::RectFlags::none);
+                }
+
+                // Draw the value label as well
+                {
+                    int16_t xPos = gs.left + gs.xOffset - 3;
+                    // int16_t width = gs.xOffset - 3; // set but not used
+                    int16_t yPos = -edx + gs.canvasHeight + gs.top - 5;
+                    if (gs.flags & (1 << 0)) // never set
+                    {
+                        yPos -= gs.canvasHeight / 2;
+                    }
+
+                    auto tr = Gfx::TextRenderer(drawingCtx);
+                    auto formatArgs = FormatArguments{};
+                    formatArgs.push(gs.yLabel);
+                    formatArgs.push<currency48_t>(ebx_eax);
+
+                    tr.drawStringRight({ xPos, yPos }, Colour::black, StringIds::graph_label_format, formatArgs);
+                }
+
+                if (gs.flags & (1 << 0)) // never set
+                {
+                    // presumably draws negative numbers as well
+                }
```

Comment:
Should we just implement this, keeping support for negative numbers? Alternatively, drop support for this flag entirely? Who knows when it might come in handy, though — perhaps for future plugins?

Reply:
i'd be more inclined to just delete it for now